kernel/src/process.rs

```diff
@@ -1197,6 +1197,7 @@ impl<C: Chip> ProcessType for Process<'a, C> {
         Some(grant_pointer)
     }
 
+    #[allow(clippy::cast_ptr_alignment)]
```

Comment:
IIRC we discussed it briefly in #1648.

There are reasons why this lint exists, and we should rather make sure that the data is indeed aligned as expected rather than ignoring the problem. In any case, this warrants at least a comment explaining why this lint is allowed.

I'd vote for fixing the problem rather than allowing this lint.

Reply:
`process.rs` currently has 4 other functions annotated with this allow, all seemingly used for casting `mem_end()` to different pointer types (`*mut *mut usize` or `*mut *mut u8` or `*const *mut u8`). 

The other allows were added in #1364, but looking back at that PR I am not sure there was any reason to believe those pointers are in fact guaranteed to be aligned? cc @bradjc

Reply:
The grant pointer ones are safe, as that's all an internal walking of an array of pointers that's guaranteed to be aligned on word-boundaries due to MPU region restrictions.

The `sp` and tbf header stuff, however, is probably not safe.

Reply:
To be honest I am not really familiar enough to know how to fix the `sp` and tbf header pointer casts to be guaranteed to be aligned (or whether those are in fact UB).

But if pat is confident that the allow added here is acceptable, perhaps we could add a comment indicating the grant pointer casts are safe, and then create an issue indicating the `sp` and tbf header pointer casts may not be?

Reply:
I added comments for the grants and stack pointer, as those are okay.

The `create` one still needs thinking about / tending to.